### DISCUSSION TITLE
payload type where one of the fields is a Query

### DISCUSSION THREAD

**Discussion by louisholley** [MichalLytek/type-graphql#829] [Q&A]
**Describe the issue**
apologies if this isn't the right category but there is nothing I could find on this in the docs so I assume it is. I want to define a payload type where one of the fields is a gql query, which would look like this if I was writing the schema directly:

```
type AddCommentPayload {
  comment: Comment!
  refetch: Query!
}
```

but not entirely sure how to achieve this with type-graphql...would I need to define a custom type to represent Query?

**Are you able to make a PR that fix this?**
wouldn't know where to start unfortunately!

**Comment by MichalLytek** [comment]
That was discussed a few times already - please use search capabilities of the GitHub repo.

Basically, it won't be supported as it's an antipattern.

**Reply by louisholley** [reply]
@MichalLytek sorry, I did try to search for it beforehand but couldn't find anything! I'd really appreciate it if you could point me in the direction of one of the other issues so I can read more on this :)

**Reply by MichalLytek** [reply]
https://github.com/MichalLytek/type-graphql/issues/427

**Reply by louisholley** [reply]
@MichalLytek fair enough! not sure if you've seen [this article](https://medium.com/@danielrearden/a-better-refetch-flow-for-apollo-client-7ff06817b052), but it provides a much better solution to refetching data affected by your mutation than apollo's built in `refetchQueries` solution. shame you won't support it, but thanks anyway

**Reply by MichalLytek** [reply]
So if I add new component with a new query related to `Comment`, I need to edit all the existing mutations to also refetch those other queries? Doesn't sound scalable nor maintainable.

If that would be documented and suggested approach from official Apollo docs I would consider supporting it. Now it's just a not so popular pattern that can be used only in some cases.
